hyp3lib/par_s1_slc_single.py

```diff
@@ -1,140 +1,98 @@
+        m = glob.glob(f'measurement/s1*-iw{swath}*')[0]
+        n = glob.glob(f'annotation/s1*-iw{swath}*')[0]
+        o = glob.glob(f'annotation/calibration/calibration-s1*-iw{swath}*')[0]
+        p = glob.glob(f'annotation/calibration/noise-s1*-iw{swath}*')[0]
     else:
-        m = glob.glob("measurement/s1*-iw{VAL}*{POL}*".format(VAL=val,POL=pol))[0]
-        n = glob.glob("annotation/s1*-iw{VAL}*{POL}*".format(VAL=val,POL=pol))[0]
-        o = glob.glob("annotation/calibration/calibration-s1*-iw{VAL}*{POL}*".format(VAL=val,POL=pol))[0]
-        p = glob.glob("annotation/calibration/noise-s1*-iw{VAL}*{POL}*".format(VAL=val,POL=pol))[0]
-    cmd = "par_S1_SLC {m} {n} {o} {p} {path}/{acq}_00{VAL}.slc.par {path}/{acq}_00{VAL}.slc {path}/{acq}_00{VAL}.tops_par".format(acq=acqdate,m=m,n=n,o=o,p=p,VAL=val,path=path) 
+        m = glob.glob(f'measurement/s1*-iw{swath}*{pol}*')[0]
+        n = glob.glob(f'annotation/s1*-iw{swath}*{pol}*')[0]
+        o = glob.glob(f'annotation/calibration/calibration-s1*-iw{swath}*{pol}*')[0]
+        p = glob.glob(f'annotation/calibration/noise-s1*-iw{swath}*{pol}*')[0]
+    
+    cmd = f'par_S1_SLC {m} {n} {o} {p} {out_dir}/{acquisition_date}_00{swath}.slc.par ' \
+          f'{out_dir}/{acquisition_date}_00{swath}.slc {out_dir}/{acquisition_date}_00{swath}.tops_par'
+    
     return cmd
 
 
-def par_s1_slc_single(myfile, pol=None):
+def par_s1_slc_single(safe_dir, pol='vv', orbit_file=None):
+    """Pre-process S1 SLC imagery into GAMMA format SLCs
 
+    Args:
+        safe_dir: Sentinel-1 SAFE directory location
+        pol: polarization (e.g., 'vv')
+        orbit_file: Orbit file to use (will download a matching orbit file if None)
+    """
     wrk = os.getcwd()
-   
-    if pol is None:
-        pol = 'vv'
-
-    logging.info("Procesing directory {}".format(myfile))
-    mytype = myfile[13:16]
-    logging.info("Found image type {}".format(mytype))
+    pol = pol.lower()
 
-    if "SSH" in mytype or "SSV" in mytype:
-         logging.info("Found single pol file")
-         single_pol = 1
-    elif "SDV" in mytype:
-         logging.info("Found multi-pol file")
-         single_pol = 0
-         if "hv" in pol or "hh" in pol:
-             logging.error("ERROR: no {} polarization exists in a {} file".format(pol,mytype))
-             exit(1)
-    elif "SDH" in mytype:
-         logging.info("Found multi-pol file")
-         single_pol = 0
-         if "vh" in pol or "vv" in pol:
-             logging.error("ERROR: no {} polarization exists in a {} file".format(pol,mytype))
-             exit(1)
+    logging.info(f'Procesing directory {safe_dir}')
+    image_type = safe_dir[13:16]
+    logging.info(f'Found image type {image_type}')
 
-    folder = myfile.replace(".SAFE","")
-    datelong = myfile.split("_")[5]
-    acqdate = (myfile.split("_")[5].split("T"))[0]
-    path = os.path.join(wrk,acqdate)
+    folder = safe_dir.replace('.SAFE', '')
```

Comment:
only being used by `os.chdir(f'{folder}.SAFE')` below, no reason to strip `.SAFE` just to add it back

Reply:
That is rather silly! good catch